src/main/java/com/google/sps/servlets/YoutubeServlet.java

```diff
@@ -14,12 +14,76 @@
 
 package com.google.sps.servlets;
 
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.api.services.youtube.YouTube;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collection;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
 
 /**
- * Servlet that fetch from Youtube Server.
+ * Servlet that fetches from Youtube Server.
  */
-@WebServlet("/userLogIn")
+@WebServlet("/YouTubeComments")
 public class YoutubeServlet extends HttpServlet {
+  private static final String URL_PARAMETER = "url";
+  private static final String SNIPPET_PARAMETER = "snippet,replies";
+  private static final String ORDER_PARAMETER = "relevance";
+  private static final Long COMMENT_LIMIT = 100L;
+  // TODO: obtain actual API Key
+  private static final String DEVELOPER_KEY = "OUR_API_KEY"; 
+  private static final String APPLICATION_NAME = "SAY";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  /**
+   * Applies parameters to the YouTube API, then extracts comments.
+   * URL is the only true variable; for this application we will always want
+   * order to be relevance and max results to be 100, the API's arbitrary limit
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      YouTube youtubeService = getService();
+      String url = request.getParameter(URL_PARAMETER);
+      // Define and execute the API request
+      YouTube.CommentThreads.List commentRequest = youtubeService.commentThreads()
+          .list(SNIPPET_PARAMETER);
+      CommentThreadListResponse commentResponse = commentRequest.setKey(DEVELOPER_KEY)
+          .setVideoId(url)
+          .setOrder(ORDER_PARAMETER)
+          .setMaxResults(COMMENT_LIMIT)
+          .execute();
+      Statistics statistics = new Statistics(new ArrayList<Double>(),0);
```

Comment:
Uri it is not on your side to create statistics. I will return you back a statistics object; what you need is to call my  method with your commentThreadListResponse

Reply:
I know. This is a placeholder so that it returns something. Once yours is merged, I will use that.